### PR TITLE
[DOCS] Replace deprecated --template option with --index-management

### PR DESCRIPTION
**Pull request by dedemorton**
Replaces remaining instances of the deprecated option (--template).

### CODE REVIEW

File: docs/devguide/newdashboards.asciidoc [hunk 1]
```diff
@@ -72,7 +72,7 @@ Flags:
   -h, --help               help for setup
       --machine-learning   Setup machine learning job configurations only
       --modules string     List of enabled modules (comma separated)
-      --template           Setup index template only
+      --index-management           Setup index template only
```
Comment (simitt) [inline docs/devguide/newdashboards.asciidoc]: Please also update the explanation to something like `Setup index management only (templates and ILM policies)`.

Comment (dedemorton) [reply docs/devguide/newdashboards.asciidoc]: Looking at this more closely, if we're going to show the output of running  `metricbeat help setup`, the example should show the exact output of running this command.

File: docs/devguide/newdashboards.asciidoc [hunk 2]
```diff
@@ -49,10 +49,12 @@ To import the dashboards, run the `setup` command.
 The `setup` phase loads:
 
 - Index mapping template in Elasticsearch
-- Kibana dashboards
-- Machine Learning jobs (if available)
+- Kibana dashboards (where available)
+- Machine Learning (ML) jobs (where available)
+- Ingest pipelines (where available)
+- ILM policy (for Elasticsearch 6.5 and newer)
```
Comment (simitt) [inline docs/devguide/newdashboards.asciidoc]: Actually, not sure about the ES version here, afaik it should be >= 6.7 (supported) or >= 7.0 (default).

Comment (dedemorton) [reply docs/devguide/newdashboards.asciidoc]: Actually I'm not sure it's necessary to mention the version number at all here.

Comment (dedemorton) [reply docs/devguide/newdashboards.asciidoc]: @simitt ^^ The version info shown here is taken directly from the output of `help setup`. I'm not sure it's necessary here, though, so Ill will change the lead-in wording in this section.

File: docs/devguide/newdashboards.asciidoc [hunk 3]
```diff
@@ -49,10 +49,12 @@ To import the dashboards, run the `setup` command.
 The `setup` phase loads:
```
Comment (dedemorton) [inline docs/devguide/newdashboards.asciidoc]:
```suggestion
The `setup` phase loads several dependencies, such as: 
```

File: docs/devguide/newdashboards.asciidoc [hunk 4]
```diff
@@ -49,10 +49,12 @@ To import the dashboards, run the `setup` command.
...
 
 - Index mapping template in Elasticsearch
-- Kibana dashboards
-- Machine Learning jobs (if available)
+- Kibana dashboards (where available)
```
Comment (dedemorton) [inline docs/devguide/newdashboards.asciidoc]:
```suggestion
- Kibana dashboards
```

File: docs/devguide/newdashboards.asciidoc [hunk 5]
```diff
@@ -49,10 +49,12 @@ To import the dashboards, run the `setup` command.
...
+- Machine Learning (ML) jobs (where available)
```
Comment (dedemorton) [inline docs/devguide/newdashboards.asciidoc]:
```suggestion
- Machine Learning (ML) jobs
```

File: docs/devguide/newdashboards.asciidoc [hunk 6]
```diff
@@ -49,10 +49,12 @@ To import the dashboards, run the `setup` command.
...
+- Ingest pipelines (where available)
```
Comment (dedemorton) [inline docs/devguide/newdashboards.asciidoc]:
```suggestion
- Ingest pipelines
```

File: docs/devguide/newdashboards.asciidoc (hunk 2 above)
Comment (dedemorton) [inline docs/devguide/newdashboards.asciidoc]:
```suggestion
- ILM policy
```

File: docs/devguide/newdashboards.asciidoc [hunk 7]
```diff
@@ -46,13 +46,15 @@ To import the dashboards, run the `setup` command.
...
 
-The `setup` phase loads:
+The `setup` phase loads several dependencies, such as: 
 
 - Index mapping template in Elasticsearch
 - Kibana dashboards
-- Machine Learning jobs (if available)
+- Machine Learning (ML) jobs
+- Ingest pipelines
+- ILM policy
```
Comment (dedemorton) [inline docs/devguide/newdashboards.asciidoc]:
```suggestion

The actual list varies depending on the Beat you're setting up.
```

File: docs/devguide/newdashboards.asciidoc [hunk 8]
```diff
@@ -46,13 +46,17 @@ To import the dashboards, run the `setup` command.
...
 
-For more details about the `setup` command, run the following:
+The dependencies vary depending on the Beat you're setting up.
+
+For more details about the `setup` command, run:
```
Comment (dedemorton) [inline docs/devguide/newdashboards.asciidoc]:
```suggestion
For more details about the `setup` command, see the command-line help. For example:
```

File: docs/devguide/newdashboards.asciidoc [hunk 9]
```diff
@@ -63,16 +67,18 @@ This command does initial setup of the environment:
  * Index mapping template in Elasticsearch to ensure fields are mapped.
  * Kibana dashboards (where available).
  * ML jobs (where available).
+ * Ingest pipelines (where available).
+ * ILM policy (for Elasticsearch 6.5 and newer).
```
Comment (dedemorton) [inline docs/devguide/newdashboards.asciidoc]: @simitt Note that this the exact output of running `metricbeat help setup` (mentions 6.5), so I don't think I should change the version here.

Comment (simitt) [reply docs/devguide/newdashboards.asciidoc]: Thanks @dedemorton, afaik it was _possible_ to import the policies from 6.5 on (beta), but one would need to set `setup.ilm.enabled: true` as with `auto` it only enables and therefore loads policies for ES >= `7.0`. I think it is ok to stick with this text, as it doesn't specify any further.

**Comment by dedemorton** [comment]
jenkins test this again please

**Pull request merged**